photon-core/src/main/java/org/photonvision/common/hardware/HardwareManager.java

```diff
@@ -165,6 +165,53 @@ public boolean restartDevice() {
         }
     }
 
+    /**
+     * Get the MAC address of the device
+     *
+     * @return The MAC address of the device, or null if it could not be determined
+     */
+    public String getMACAddress() {
+        if (Platform.isLinux()) {
+            try {
+                // Try using ip link to get MAC addresses
+                String command =
+                        "ip link | grep -A1 'state UP' | grep -i 'link/ether' | head -n1 | awk '{print $2}'";
```

Comment:
If device networking is managed, we should already know the name of the interface right? Will that simplify this code?